go/consensus/tendermint/abci/mux.go

```diff
@@ -859,6 +869,17 @@ func (mux *abciMux) checkDependencies() error {
 	return nil
 }
 
+func (mux *abciMux) GetApplicationState() *ApplicationState {
+	return mux.state
+}
+
+func NewTestMuxer(ctx context.Context, cfg *ApplicationConfig) (*abciMux, func(), error) { // nolint
```

Comment:
Hm so this exported function returns a private type (`abciMux`), how does this work?